src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -134,16 +127,28 @@ public class SqlMutationEngine : IMutationEngine
                                 sqlMetadataProvider);
 
                         // If the number of records affected by DELETE were zero,
-                        // and yet the result was not null previously, it indicates this DELETE lost
-                        // a concurrent request race. Hence, empty the non-null result.
                         if (resultProperties is not null
                             && resultProperties.TryGetValue(nameof(DbDataReader.RecordsAffected), out object? value)
-                            && Convert.ToInt32(value) == 0
-                            && result is not null && result.Item1 is not null)
+                            && Convert.ToInt32(value) == 0)
+                        {
+                            // the result was not null previously, it indicates this DELETE lost
+                            // a concurrent request race. Hence, empty the non-null result.
+                            if (result is not null && result.Item1 is not null)
+                            {
+
+                                result = new Tuple<JsonDocument?, IMetadata?>(
+                                    default(JsonDocument),
+                                    PaginationMetadata.MakeEmptyPaginationMetadata());
+                            }
+                            else if (context.Selection.Type.TypeName() == GraphQLUtils.DB_OPERATION_RESULT_TYPE)
```

Comment:
do we want the detection of `GraphQLUtils.DB_OPERATION_RESULT_TYPE` as a typename to be used outside of DW?

Reply:
In the design discussion, we had discussed this as a possibility that people could opt into. Sample scenarios: for Bulk inserts, bulk deletes etc user may not want to see the records returned but instead just want a summary represented by the db_operation_result type. We can expose this in the runtimeconfig. Similar thing can be used by users of other datasources if they just want summary results as opposed to modified records. Hence there is scope for this to be used for other data sources in the future.